### PR TITLE
feature (new integration): Onboard Monday source

### PR DESCRIPTION
**Pull request by yashasvibajpai**
## Description of the change

>  Onboard Monday source

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/sources/monday/transform.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+// the payload for the challenge event will be as follows:
+// {
+//  challenge : "some_key"
+// }
+// this will be sent when the webhook is added to an item in monday.
+
+function isChallengeEvent(event) {
+  if (event?.challenge) return true;
+  return false;
```
Comment (ItsSudip) [inline v0/sources/monday/transform.js:53]:
```suggestion
  return !!event?.challenge;
```

File: v0/sources/monday/transform.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+// we will check here if the event is a challenge event or not
+// and process accordingly.
+// For challenge event the recieved challenge object is sent back
+// to Monday to verify the webhook url.
+// Ref: https://developer.monday.com/api-reference/docs/webhooks-1#how-to-verify-a-webhook-url
+function process(event) {
+  const response = isChallengeEvent(event)
+    ? processChallengeEvent(event)
+    : processEvent(event);
```
Comment (ItsSudip) [inline v0/sources/monday/transform.js:75]: Don't use the ternary operator here.

**Review by ItsSudip — CHANGES_REQUESTED** [review]

File: v0/sources/monday/transform.js (hunk 1 above)
Comment (yashasvibajpai) [reply v0/sources/monday/transform.js:53]: Updated.

File: v0/sources/monday/transform.js (hunk 2 above)
Comment (yashasvibajpai) [reply v0/sources/monday/transform.js:74]: This can be used here.

**Review by ItsSudip — CHANGES_REQUESTED** [review]

File: v0/sources/monday/transform.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
```
[the remark this answers is not in the document]
Comment (yashasvibajpai) [reply v0/sources/monday/transform.js]: updated!

**Review by ItsSudip — DISMISSED** [review]

**Review by utsabc — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

**Pull request merged**